common/txmgr/txmgr.go

```diff
@@ -348,20 +350,20 @@ func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) runLoop()
 		wg.Add(2)
 		go func() {
 			defer wg.Done()
-			ctx, cancel := b.chStop.NewCtx()
+			ctx2, cancel := b.chStop.NewCtx()
```

Comment:
Let's pass `ctx` through the function signature instead:
```suggestion
```
`execReset(ctx, nil)`

Reply:
Thanks - great call. The go routine to start the confirmer works that way already and is more straightforward because it listens for `b.chStop` instead of creating a context from chStop and listening for the Done signal on that context